tools/java/org/unicode/cldr/util/XPathParts.java

```diff
@@ -1257,8 +1285,12 @@ public XPathParts cloneAsThawed() {
          * Reference: https://unicode.org/cldr/trac/ticket/12007
          */
         xppClone.dtdData = this.dtdData;
-        for (Element e : this.elements) {
-            xppClone.elements.add(e.cloneAsThawed());
+        if (!frozen) {
```

Comment:
X’ = X.cloneAsThawed(), two cases:
If X is not frozen, X’ should be a deep copy.
If X is frozen, we can just make a shallow copy X'.